Stemma for Njals saga

cmsmcq opened the pull request:
This pull request adds a directory for Old Norse ("non") and a stemma for Njáls saga as constructed by Einar Ólafur Sveinsson in his 1954 edition.

I have tried to follow the submission guidelines but am not sure I got everything right; if things should be done differently, I would be grateful if you would tell me what to do differently.

I do not currently have a scan of the stemma in Einar's edition but should be able to supply it.

GusRiva commented:
Thanks for the contribution! It looks great and a good way to start the non folder!

If you can submit the image of the stemma that would make it easier to check. 

There is one thing that could be improved: When many manuscripts are grouped together as being descendants of the same ancestor, we need to still create a node for each of them – even if the printed stemma justs grops them. For example, this:

Xstar -> K2R2R4O3 

should be: 

Xstar -> K2
Xstar -> R2
Xstar -> R4
Xstar -> O3

Otherwise, a program parsing the DOT-file would interpret it as there being one textual witness there, when actually there are many.

cmsmcq commented:
Thank you!

I have now added PNG images of the stemma, which Einar divides into two parts for typographic reasons.

I have changed stemma.gv to eliminate the groups.   Is there a convention to be used in Graphviz to distinguish the case where the stemma clearly says "Each of these manuscripts is independently descended from their common parent" from the case where the stemma says "These manuscripts form a family whose internal structure and relations have not been clarified"?   For some purposes it will make a difference, I think.

For the moment I have made the arcs gray, although that might be interpreted to mean that it's uncertain whether the descendants are attached to the correct ancestor (which is not the case here).  

I've also re-read the editor's discussion of the manuscripts which have more than one exemplar, and marked O2 as a case of contamination.  Since Einar seems confident that the portions of Bb, Ga, K, O (other than O2), S, Sv, and R can be treated as separate units each with a single exemplar, I have *not* coded those mss as cases of contamination.

GusRiva commented:
Thanks for the changes and sorry for the late response.

> I have now added PNG images of the stemma, which Einar divides into two parts for typographic reasons.
Just for the purpose of validation and internal structural coherence, I will combine the two images into a single file.

> Is there a convention to be used in Graphviz to distinguish the case where the stemma clearly says "Each of these manuscripts is independently descended from their common parent" from the case where the stemma says "These manuscripts form a family whose internal structure and relations have not been clarified"? For some purposes it will make a difference, I think.

Yes, that is a valid point, but we don't currently have a solid method to encode that difference. We can of course always add a note, that is a prose comment, but that is of course not machine readable. For now we will have to live with this. 

> For the moment I have made the arcs gray, although that might be interpreted to mean that it's uncertain whether the descendants are attached to the correct ancestor (which is not the case here).

Yes, that is currently confusing. I will remove the color.

Jean-Baptiste-Camps commented:
Hello ! And many thanks for this contribution !

Perhaps I would advocate keeping the gray: the relation are indeed unclear, or underspecified, which does not seem too far from the semantics of the gray links, I think.

For the images, perhaps I can do it after the merge. That is also a first.

Ah, each stemma brings its own questions !

Jean-Baptiste-Camps reviewed (approved):
Let's merge and fix the pics !
